lib/uptime/env.ex

```diff
@@ -27,23 +31,81 @@ defmodule Uptime.Env do
     :ok
   end
 
+  @doc """
+  Read a value from the environment.
+  First check for a secrets file, then the environment, then the config file.
+  secrets file > environment variable > config file
+  """
   @spec read(String.t()) :: String.t()
   def read(name) do
-    case File.read("/etc/secrets/#{name}") do
-      {:ok, value} -> value
-      _not_found -> System.get_env(name)
+    prefixed_name = "#{@prefix}_#{name}"
+
+    secret_file_result =
+      @secrets_directory
+      |> Path.join(prefixed_name)
+      |> File.read()
+
+    case secret_file_result do
+      {:ok, value} ->
+        value
+
+      _not_found ->
+        case read_from_env(prefixed_name) do
+          {:ok, value} ->
+            value
+
+          _not_found ->
+            name
+            |> String.downcase()
+            |> read_from_config_file(find_config_path())
+        end
     end
   end
 
+  @spec read_from_env(String.t()) :: {:ok, String.t()} | :error
+  defp read_from_env(name) do
+    value = System.get_env(name)
+
+    if valid_env_var_value(value) do
+      {:ok, value}
+    else
+      :error
+    end
+  end
+
+  @spec read_from_config_file(String.t(), String.t()) :: any() | nil
+  defp read_from_config_file(key, path) when is_binary(path) do
+    case YamlElixir.read_from_file(path) do
```

Comment:
awesome, would we also want to support json config?

Reply:
Ya I think we could no problem. Let me add that.

Reply:
Wait let's add it later if we find a need